unity-client/Assets/Scripts/MainScripts/DCL/TheGraph/TheGraph.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using DCL.Helpers;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public enum TheGraphCache
+{
+    DontUseCache,
+    UseCache
+}
+
+public interface ITheGraph
+{
+    Promise<string> Query(string url, string query);
+    Promise<string> Query(string url, string query, QueryVariablesBase variables);
+    Promise<List<Land>> QueryLands(string tld, string address, TheGraphCache cache);
+}
+
+public class TheGraph : ITheGraph
+{
+    private const float CACHE_TIME = 5 * 60;
+
+    private readonly Dictionary<string, QueryLandCache> landQueryCache = new Dictionary<string, QueryLandCache>();
+
+    public Promise<string> Query(string url, string query) { return Query(url, query, null); }
+
+    public Promise<string> Query(string url, string query, QueryVariablesBase variables)
+    {
+        Promise<string> promise = new Promise<string>();
+
+        if (string.IsNullOrEmpty(query) || string.IsNullOrEmpty(url))
+        {
+            promise.Reject($"error: {(string.IsNullOrEmpty(url) ? "url" : "query")} is empty");
+            return promise;
+        }
+
+        string queryJson = $"\"query\":\"{Regex.Replace(query, @"\p{C}+", string.Empty)}\"";
+        string variablesJson = variables != null ? $",\"variables\":{JsonUtility.ToJson(variables)}" : string.Empty;
+        string bodyString = $"{{{queryJson}{variablesJson}}}";
+
+        var request = new UnityWebRequest();
+        request.url = url;
+        request.method = UnityWebRequest.kHttpVerbPOST;
```

Comment:
Just out of curiosity: Is this kind of method `kHttpVerbPOST` a POST requests but it is also returning data as a GET method?

Reply:
Do you think it would be worth using the new WebRequest system with retries here? (when this PR be merged: https://github.com/decentraland/explorer/pull/2256) In this case we would have to modify it to allow request methods like `kHttpVerbPOST`.

Reply:
Yeap, it's as you suspected :P
I don't know if we can use the new WebRequest, we can try, but as you can see I had to do a lot of sorcery there to make it work with The Graph